lib/openapi_parser/schema_validators/all_of_validator.rb

```diff
@@ -7,6 +7,9 @@ class AllOfValidator < Base
     def coerce_and_validate(value, schema)
       # if any schema return error, it's not valida all of value
       schema.all_of.each do |s|
+        # We need to store the reference to all of, so we can perform strict check on allowed properties
+        s.parent_all_of = schema.all_of
```

Comment:
I think Schema object should be immutable.
And schema's parent isn't always the same (e.g, when we use reference, one object have many parent)
so it becomes a problem when multi-threading.

Reply:
NotExistPropertyDefinition error have property name which not defined in `s`.
I think this validator validate all schemas in all_of section and collect not defined keys.
And returns error if there is property name not included in all the definitions.

This is sample code
```ruby
remain_keys = value.keys
schema.all_of.each do |s|
  _coerced, errs = validatable.validate_schema(value, s, early_return=false)
  errors_key = get_errors_key(errs)
  remain_keys = remain_keys & errors_key
end

raise 'not exist key' if remain_keys
```

Reply:
But this sample code dosen't work because validate_schema method return only one error.
So I think we should add early_return option which return all validate errors.

Reply:
right, doing it here with `schema.all_of.each` would solve only a part of the issue. Because there is a normal object validation, which changes if the object is under `allOf` https://github.com/ota42y/openapi_parser/pull/33/files#diff-0662b153c9d300d215ae97087f9fa776R18

So I can do the validation here, but I will still need to pass something to this nested validation, so we can skip it at least. So maybe passing another argument to `validatable.validate_schema(value, s)` ?

Reply:
so I moved the parent_all_of to argument https://github.com/ota42y/openapi_parser/pull/33/commits/fde88c0f780054cdadd9aace5f4f6a26cf4bb691 and now trying to figure out how to move it up and aggregate, but still it won;t be perfect, because it can throw also OpenAPIParser::NotExistRequiredKey and we can't aggregate 2 different errors into one.

Reply:
@ota42y  ok, it's done